### PR TITLE
fix(protocol-engine): Fix hanging after cancellation

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

Fixes RQA-2091

# Test Plan

* [x] Flex and OT-2 protocols can still succeed
* [x] Flex protocols, when cancelled, home the X/Y/Z axes but leave the plungers and tip ejectors alone
* [x] OT-2 protocols, when cancelled, eject the tips in the fixed trash if there are any tips attached

# Changelog

* After a protocol ends, *always* call `hardware_api.stop` before doing anything further with the hardware. This is the bug fix.
* Formerly, if a protocol ended without any tips attached, we were taking the opportunity to home the plungers. That's dropped in this PR; for simplicity, we will never ever home the plungers as part of run cleanup. Homing the plungers is taken care of at run *start.* @sfoster1 and I agree that there was no benefit to homing the plungers during run cleanup, and that it could only open up opportunities for something to go wrong.

# Review requests

There are opportunities for further cleanup here. For example, `HardwareStopper._drop_tip()` is *also* stopping and homing, which makes the overall sequence hard to follow and seems outside of its responsibility.

Since this PR has already been tested *without* that cleanup, let's not do it in this PR—but if you have any ideas, please do leave comments.

# Risk assessment

Medium. I haven't looked at how well this code is covered by unit tests, but this PR is passing CI without having had to change anything, which suggests there are gaps.

-----

Co-authored with @jbleon95 and @sfoster1

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Looks good to me if it works, I think this is the right approach for homing

**Review by jbleon95 — APPROVED** [review]
Fixes the cancellation bug and looks correct to me
